### PR TITLE
Feature/komarch 37 posts frontend

### PR DESCRIPTION
**Pull request by igor-kamil**
## list of posts
screenshot-new komarch local-2021-02-03-00-57-06-340

## post detail
screenshot-new komarch local-2021-02-03-00-58-27-882

### CODE REVIEW

File: resources/js/components/filters.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+$(document).ready(function(){
+    $("#filters :input").change(function(e) {
```
Comment (yanchith) [inline resources/js/components/filters.js:2]: Does this work for you? Changing filters on page doesn't seem to do anything for me on any browser.

File: resources/views/components/filter-checkbox.blade.php [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div class="form-check pb-1">
+    <input class="form-check-input" type="checkbox" value="{{ $tag->name }}" id="{{ $tag->slug }}" name="categories[]" {{ (in_array($tag->name, request('categories', [])) ? 'checked' : '' ) }}>
```
Comment (yanchith) [inline resources/views/components/filter-checkbox.blade.php:2]: Asking from the education perspective... So I've seen this pattern around the codebase where a UI component directly accesses a variable from a request like this one (`request('categories')`). Is this common practice in Laravel land? Seems to me like that this is very prone to breakage. Should the UI components really know about request variables?

File: resources/views/components/tags.blade.php [hunk 3]
```diff
@@ -1,7 +1,7 @@
 @if($tags)
     <div class="tags">
         @foreach ($tags->sortBy->name as $tag)
-            <a class="btn btn-outline-dark btn-sm mb-2" href="{{ route('posts.index', ['filters[]' => $tag->slug]) }}" role="button">{{ $tag->name }}</a>
+            <a class="btn btn-outline-dark btn-sm mb-2 small" href="{{ route('posts.index', ['categories[]' => $tag->name]) }}" role="button">{{ $tag->name }}</a>
```
Comment (yanchith) [inline resources/views/components/tags.blade.php:4]: Why the change from `slug` to `name`?

**Review by yanchith — COMMENTED** [review]
Design looks nice! I do have some questions inline though.

File: resources/js/components/filters.js (hunk 1 above)
Comment (igor-kamil) [reply resources/js/components/filters.js:2]: yes, it works for me. did you re-compiled you assets? `npm run dev` or `npm run watch` (my preference is the 2nd option)

File: resources/views/components/filter-checkbox.blade.php (hunk 2 above)
Comment (igor-kamil) [reply resources/views/components/filter-checkbox.blade.php:2]: I wouldn't call it best pattern. 
However I don't see any problem from my perspective... Before I used to sanitise request data in controller and pass to view as `$active_categories` or something like that... 
Can you be more specific, what might get broken?

File: resources/views/components/tags.blade.php (hunk 3 above)
Comment (igor-kamil) [reply resources/views/components/tags.blade.php:4]: because here https://github.com/SlovakNationalGallery/komarch.sk/pull/18/files/35d8a2836446074bab30cfb7c8780ae2d70c9aad#diff-07a440539f1019ca1d1a6ba20b1bf7afc976f594df9ae99b99fb84ccffc53b14R17 it expect raw tag value (if I understand it correctly) - and it works

File: resources/views/components/filter-checkbox.blade.php (hunk 2 above)
Comment (yanchith) [reply resources/views/components/filter-checkbox.blade.php:2]: Hahaha, yeah. Your previous approach is what I would do. This way components (and even views) are independent of the route they are called from. One more upgrade would be to have the thing typed, but I don't think that is possible in laravel views. Or can they have typed params?

File: resources/js/components/filters.js (hunk 1 above)
Comment (yanchith) [reply resources/js/components/filters.js:2]: Damn... worked on Rust meanwhile and forgot that I have to build everything 3 times in PHP. My bad.

Comment (yanchith) [reply resources/js/components/filters.js:2]: Hmm, okay, now it updates, but if I select any category, I get zero results. Before I dive in to debug this, does that work for you?

Comment (igor-kamil) [reply resources/js/components/filters.js:2]: 😅 I thought it's really broken. it took me few minutes to see what happened. 
the problem is in the current locale. 
we have `'useAcceptLanguageHeader' => true` in `config/laravellocalization.php`
and in english locale it doesn't work
I will switch it to `false`... since there are no reasonable translations anyway
the proper and long-term solution is to implement filter by tags using elasticsearch. that will happen later. 
this was supposed to be frontend task and I "made it working" from my own willingness and impatience  :D

File: resources/views/components/filter-checkbox.blade.php (hunk 2 above)
Comment (igor-kamil) [reply resources/views/components/filter-checkbox.blade.php:2]: well. to be honest ... I reused the code that we already wrote in regarch/webumenia. it's proofed to work. 
on the other hand I admit you are principally right. 
still I think **done is better then perfect**
